lib/src/native/realm_core.dart

```diff
@@ -393,6 +395,17 @@ class _RealmCore {
     return config.shouldCompactCallback!(totalSize, usedSize) ? TRUE : FALSE;
   }
 
+  static void _syncErrorHandlerCallback(Pointer<Void> userdata, Pointer<realm_sync_session> user, realm_sync_error error) {
+    final FlexibleSyncConfiguration? syncConfig = userdata.toObject(isPersistent: true);
```

Comment:
we should be getting the error handler directly here or nullptr if there was no handler registered

Reply:
OK.

Reply:
I need syncConfig, because depends on is_client_reset_requested we caldifferent event handlers. See #608

Reply:
I see